diff --git a/public/common/TracySystem.cpp b/public/common/TracySystem.cpp
--- a/public/common/TracySystem.cpp
+++ b/public/common/TracySystem.cpp
@@ -93,6 +93,9 @@ uint32_t GetThreadHandleImpl()
 #elif defined __EMSCRIPTEN__
     // Not supported, but let it compile.
     return 0;
+#elif defined __SWITCH__
+    // Not supported, but let it compile.
+    return 0;
 #else
     // To add support for a platform, retrieve and return the kernel thread identifier here.
     //
@@ -353,9 +356,11 @@ TRACY_API const char* GetUserLogin()
 #elif defined __ANDROID__
     const auto user = getlogin();
     if( user ) return user;
     return "(?)";
+#elif defined __SWITCH__
+    return "switch";
 #else
     static char user[1024] = {};
     getlogin_r( user, sizeof( user ) );
     return user;
 #endif
 }
